src/lib/ast/skillsRelation.ts

```diff
@@ -0,0 +1,19 @@
+ 
+import { Skill } from "../types";
+
+export class SkillsRelations{
+	private parents: ReadonlyArray<Skill>;
+
+	constructor(public skills: ReadonlyArray<Skill>) {
+		this.parents = skills.filter(skill => skill.nestedSkills.length > 0);
+	}
+
+	getChildren(parent: Skill): Skill[] {

```

Comment:
Seems to have a poor performance as you do not use the `parents`property of the class. Instead you always iterate over all skills and check if they are the parent of the skill to be searched for...

The idea of such a class is to reduce the amount of loop-calls while evaluating all the expressions.